src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -85,10 +87,10 @@ private static void HandleStartObject(JsonSerializerOptions options, ref ReadSta
                 // Nested preserved array within another preserved array.
                 Debug.Assert(options.ReferenceHandling.ShouldReadPreservedReferences());
 
-                Type preservedObjType = state.Current.JsonPropertyInfo.GetJsonPreservedReferenceType();
+                Type preservedObjType = state.Current.JsonPropertyInfo!.GetJsonPreservedReferenceType();
                 // Re-Initialize the current frame.
                 state.Current.Initialize(preservedObjType, options);
-                state.Current.ReturnValue = state.Current.JsonClassInfo.CreateObject();
+                state.Current.ReturnValue = state.Current.JsonClassInfo!.CreateObject!();
```

Comment:
We can use nullable forgiving `!` on `CreateObject` safely.
The reason for that is that at this point we will always return the `JsonClassInfo` for `JsonPreservedReference<T>` which essentially never contains `CreateObject` with a `null` value.

@ahsonkhan

Reply:
Please add that as a comment in source.